server/src/main/java/org/elasticsearch/cluster/coordination/ClusterFormationFailureHelper.java

```diff
@@ -177,36 +190,35 @@ public record ClusterFormationState(
 
         public ClusterFormationState(
             Settings settings,
-            ClusterState clusterState,
+            ClusterFormationClusterStateView clusterFormationClusterStateView,
             List<TransportAddress> resolvedAddresses,
             List<DiscoveryNode> foundPeers,
             Set<DiscoveryNode> mastersOfPeers,
-            long currentTerm,
             ElectionStrategy electionStrategy,
             StatusInfo statusInfo,
             List<JoinStatus> inFlightJoinStatuses
         ) {
             this(
                 INITIAL_MASTER_NODES_SETTING.get(settings),
-                clusterState.nodes().getLocalNode(),
-                clusterState.nodes().getMasterNodes(),
-                clusterState.version(),
-                clusterState.term(),
-                clusterState.getLastAcceptedConfiguration(),
-                clusterState.getLastCommittedConfiguration(),
+                clusterFormationClusterStateView.localNode,
+                clusterFormationClusterStateView.masterEligibleNodes,
+                clusterFormationClusterStateView.lastAcceptedVersion,
+                clusterFormationClusterStateView.lastAcceptedTerm,
+                clusterFormationClusterStateView.lastAcceptedConfiguration,
+                clusterFormationClusterStateView.lastCommittedConfiguration,
```

Comment:
Maybe just pass in `clusterFormationClusterStateView` itself, no real need to deconstruct it like this I reckon.